### ISSUE TITLE
`copy` with matrix argument gives error

### ISSUE THREAD

**Issue by matthias314** [Nemocas/Nemo.jl#1337]
The following code works for `AbstractAlgebra`, but not for `Nemo` (v0.32.2):
```
julia> a = zero_matrix(ZZ, 2, 2);
julia> copy(a)
ERROR: MethodError: no method matching copy(::fmpz_mat)
```
The same happens with `QQ` and `GF(2)` instead of `ZZ`. Shouldn't `Nemo` provide a method for `copy`?

**Comment by thofma** [comment]
It is impossible to define a `copy` method for Nemo matrices with the same semantics. We should probably remove the `copy` method in AbstractAlgebra to make it more uniform.

**Comment by matthias314** [comment]
Maybe the exact semantics are impossible. But it must be possible to produce a matrix `B` that is equal to a given matrix `A` and that one can modify without changing `A`. I think there are use cases for such a function, and it would make sense to call it `copy`.

**Comment by thofma** [comment]
There is  `deepcopy`.

**Comment by matthias314** [comment]
Yes. Instead of removing `copy` from AbstractAlgebra, would it make sense to make it call `deepcopy` in Nemo? Plain Julia also provides `copy` for arrays, so people might expect it to exist for other matrix types. I would guess that most users are not concerned about the exact semantics, but only want a cheap way to get a copy.
